### PR TITLE
Clean up cmake organisation

### PR DESCRIPTION
**Pull request by JasonMarechal25**
- Clean up CMake organisation. Place headers in include directories
- Include directories follow the convention <project-name/inner/directory/header.h>
- Remove use of "get properties" and use target_include/target_link CMake convention

### CODE REVIEW

File: src/cpp/benders/benders_by_batch/BatchCollection.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#include "BatchCollection.h"
+#include "antares-xpansion/benders/benders_by_batch/BatchCollection.h"
```
Comment (a-zakir) [inline src/cpp/benders/benders_by_batch/BatchCollection.cpp:1]: why are you doing this?

Comment (a-zakir) [reply src/cpp/benders/benders_by_batch/BatchCollection.cpp:1]: it's adding complexity we really don't need in this project.

File: tests/cpp/lp_namer/VariableFileReaderTest.cpp [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #include <fstream>
 
 #include "LoggerBuilder.h"
```
Comment (flomnes) [inline tests/cpp/lp_namer/VariableFileReaderTest.cpp:3]: You forgot this header

File: tests/cpp/lp_namer/StudyUpdateTest.cc [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 
-#include "ActiveLinks.h"
-#include "CandidatesINIReader.h"
-#include "LinkProfileReader.h"
-#include "LinkdataRecord.h"
+#include "antares-xpansion/lpnamer/model/ActiveLinks.h"
+#include "antares-xpansion/lpnamer/input_reader/CandidatesINIReader.h"
+#include "antares-xpansion/lpnamer/input_reader/LinkProfileReader.h"
+#include "antares-xpansion/lpnamer/problem_modifier/LinkdataRecord.h"
 #include "LoggerBuilder.h"
```
Comment (flomnes) [inline tests/cpp/lp_namer/StudyUpdateTest.cc:8]: You forgot this header

File: tests/cpp/logger/logger_test.cpp [hunk 4]
```diff
@@ -3,14 +3,14 @@
 #include <ostream>
 #include <sstream>
 
-#include "BendersMathLogger.h"
-#include "ILogger.h"
+#include "antares-xpansion/benders/benders_core/BendersMathLogger.h"
+#include "antares-xpansion/xpansion_interfaces/ILogger.h"
 #include "LogPrefixManip.h"
 #include "RandomDirGenerator.h"
```
Comment (flomnes) [inline tests/cpp/logger/logger_test.cpp:9]: Forgotten ?

File: tests/cpp/benders/benders_sequential_test.cpp [hunk 5]
```diff
@@ -1,8 +1,8 @@
 #include <algorithm>
 
-#include "ArchiveWriter.h"
-#include "BendersSequential.h"
-#include "JsonWriter.h"
+#include "antares-xpansion/helpers/ArchiveWriter.h"
+#include "antares-xpansion/benders/benders_sequential/BendersSequential.h"
+#include "antares-xpansion/benders/output/JsonWriter.h"
 #include "LoggerStub.h"
 #include "RandomDirGenerator.h"
```
Comment (flomnes) [inline tests/cpp/benders/benders_sequential_test.cpp:7]: Forgotten ?

File: src/cpp/multisolver_interface/SolverFactory.cpp [hunk 6]
```diff
@@ -1,13 +1,13 @@
 
 #include "SolverXpress.h"
```
Comment (flomnes) [inline src/cpp/multisolver_interface/SolverFactory.cpp:2]: Forgotten ?

File: src/cpp/multisolver_interface/SolverFactory.cpp [hunk 7]
```diff
@@ -1,13 +1,13 @@
...
-#include "multisolver_interface/environment.h"
+#include "antares-xpansion/multisolver_interface/environment.h"
 
 #ifdef COIN_OR
 #include "SolverCbc.h"
 #include "SolverClp.h"
```
Comment (flomnes) [inline src/cpp/multisolver_interface/SolverFactory.cpp:7]: Forgotten ?

File: src/cpp/lpnamer/main/ProblemGeneration.cpp [hunk 8]
```diff
@@ -1,34 +1,35 @@
...
 #include "Version.h"
-#include "WeightsFileReader.h"
-#include "WeightsFileWriter.h"
-#include "XpansionProblemsFromAntaresProvider.h"
-#include "ZipProblemsProviderAdapter.h"
+#include "antares-xpansion/lpnamer/input_reader/WeightsFileReader.h"
+#include "antares-xpansion/lpnamer/input_reader/WeightsFileWriter.h"
+#include "antares-xpansion/lpnamer/problem_modifier/XpansionProblemsFromAntaresProvider.h"
+#include "antares-xpansion/lpnamer/problem_modifier/ZipProblemsProviderAdapter.h"
 #include "config.h"
```
Comment (flomnes) [inline src/cpp/lpnamer/main/ProblemGeneration.cpp:33]: Now, that's a lot of headers

File: src/cpp/benders/benders_mpi/include/antares-xpansion/benders/benders_mpi/BendersMpiOuterLoop.h [hunk 9]
```diff
@@ -1,6 +1,6 @@
 #pragma once
 #include "BendersMPI.h"
```
Comment (flomnes) [inline src/cpp/benders/benders_mpi/include/antares-xpansion/benders/benders_mpi/BendersMpiOuterLoop.h:2]: Forgotten ?

File: src/cpp/benders/benders_core/include/antares-xpansion/benders/benders_core/MasterUpdate.h [hunk 10]
```diff
@@ -1,6 +1,6 @@
 #pragma once
 #include "BendersBase.h"
```
Comment (flomnes) [inline src/cpp/benders/benders_core/include/antares-xpansion/benders/benders_core/MasterUpdate.h:2]: Forgotten ?

**Review by flomnes — COMMENTED** [review]
Minor remarks
- You forgot a few headers
- You might want to automatically re-order header inclusions : 
 1. STL headers
 2. antares-xpansion headers
clang-format allows this, and we use it in Simulator.
- Inclusion guards / `#pragma once` : in some headers you use `#ifndef ...#endif`, in some others you use `#pragma once`. I suggest you switch completely to `#pragma once`.

**Review by flomnes — APPROVED** [review]

**Comment by a-zakir** [comment]
**⚠️ Je ne suis pas d’accord avec ce travail, je ne vois aucun intérêt à importer cette « convention » dans ce projet. ⚠️**

**Review by a-zakir — APPROVED** [review]

**Pull request merged**